run.py

```diff
@@ -481,6 +482,40 @@ def main():
     elif args.command == 'metricsDeltaToMasterPipeline':
         nonbib_delta_to_master_pipeline(metrics_db_engine, args.metricsSchemaName, args.rowViewSchemaName, int(args.batchSize))
 
+    elif args.command == 'metricsCompare':
+        # compare the values in two metrics postgres tables
+        # useful to compare results from new pipeline to produciton pipeline
+        # read metrics records from both databases and compare
+        metrics_logger = setup_logging('metricsCompare', 'INFO')
+        metrics1 = metrics.Metrics(args.metricsSchemaName)
+        Session = sessionmaker(bind=metrics_db_engine)
+        session = Session()
+        if args.metricsSchemaName:
+            session.execute('set search_path to {}'.format(args.metricsSchemaName))
+
+        metrics2 = metrics.Metrics(args.metricsSchemaName2)
+        metrics_connection_string2 = config.get('METRICS_DATABASE2',
+                                               'postgresql://postgres@localhost:5432/postgres')
+        metrics_db_engine2 = create_engine(metrics_connection_string2)
+        Session2 = sessionmaker(bind=metrics_db_engine2)
+        session2 = Session2()
+        if args.metricsSchemaName2:
+            session2.execute('set search_path to {}'.format(args.metricsSchemaName2))
```

Comment:
since you moved to ORM for the execution of queries, the 'set search' may be not needed (and it would be good to remove if that'sthe case)

Reply:
since the code deals with multiple schemas, I believe the setting the search path is needed.

Reply:
if 'METRICS_DATABASE' and 'METRICS_DATABASE2' are the same strings; then it is needed (but in that case it doesn't make sense to have two config variables each pointing to the same database)

if they are different databases, then 'set search_path' is needed only if both have multiple schemas (which seems an odd use case; like: compare *this* installation of metrics db against *that* installation of metrics db)

so which one is it here?

also see: https://stackoverflow.com/a/32689664

Reply:
during the validation operation both METRICS_DATABASE and METRICS_DATABASE2 are used.  typically, I give them real values in local_config.py.  
the metrics schema name (like the schema for the nonbib tables) can be passed in as a command line argument.  for fast comparison, the nonbib tables need to be in separate schemas in a single postgres instance.  the is no use case where the metrics table needs so much flexibility, instead it follows the template used by the nonbib code.